microsoft/testsuites/core/timesync.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import PurePosixPath
+
+from assertpy import assert_that
+
+from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Windows
+from lisa.testsuite import simple_requirement
+from lisa.tools import Cat, Dmesg
+
+
+@TestSuiteMetadata(
+    area="time",
+    category="functional",
+    description="""
+    This test suite is related with time sync.
+    """,
+)
+class TimeSync(TestSuite):
+    ptp_registered_msg = "PTP clock support registered"
+    hyperv_ptp_udev_rule = "ptp_hyperv"
+    chrony_path = ["/etc/chrony.conf", "/etc/chrony/chrony.conf"]
+
+    @TestCaseMetadata(
+        description="""
+        https://docs.microsoft.com/en-us/azure/virtual-machines/linux/time-sync#check-for-ptp-clock-source # noqa: E501
+        This test is to check -
+            1. PTP time source is available on Azure guests (newer versions of Linux).
+            2. PTP device name is hyperv.
+            3. When accelerated network is enabled, muiltple PTP devices will
+             be available, the names of ptp are changeable, create the symlink
+             /dev/ptp_hyperv to whichever /dev/ptp entry corresponds to the Azure host.
+            4. Chrony should be configured to use this symlink instead of
+             /dev/ptp0 or /dev/ptp1.
+        """,
+        priority=2,
+        requirement=simple_requirement(unsupported_os=[Windows]),
+    )
+    def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
+        # 1. PTP time source is available on Azure guests (newer versions of Linux).
+        dmesg = node.tools[Dmesg]
+        assert_that(dmesg.get_output()).contains(self.ptp_registered_msg)
+
+        # 2. PTP device name is hyperv.
+        cat = node.tools[Cat]
+        clock_name_result = cat.run("/sys/class/ptp/ptp0/clock_name")
+        assert_that(clock_name_result.stdout).described_as(
+            f"ptp clock name is {clock_name_result.stdout}, expected name is hyperv"
```

Comment:
add a quote to name 'hyperv'. Should it say the name of '/sys/class/ptp/ptp0/clock_name' must be 'hyperv'? And it's better to give a reference document.

Reply:
https://github.com/microsoft/lisa/pull/1316/files#diff-437c5500bc74cc462be9463738c9dd4659dcfba1580452c5f0fb55f70d69d5bfR25 give the refer doc in case metadata

Reply:
The first place to explain a failed case is one line log, next is the whole log, and then source code, repro... So make sure the error message can explain why it's wrong itself.